src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,296 @@
+            var thumbPrintFiles = new[]
+            {
+                Path.Combine(directoryPath, "Microsoft.AspNetCore.StaticAssets.xml"),
+                Path.Combine(directoryPath, "Microsoft.AspNetCore.StaticAssets.cache"),
+            };
+
+            foreach (var file in thumbPrintFiles)
+            {
+                var thumbprint = GetThumbPrint(file);
+                thumbPrints[file] = thumbprint;
+            }
+
+            // Act
+            var incremental = await DotnetMSBuild("GenerateAspNetCoreStaticAssetsManifest");
+
+            // Assert
+            Assert.BuildPassed(incremental);
+
+            foreach (var file in thumbPrintFiles)
+            {
+                var thumbprint = GetThumbPrint(file);
+                Assert.Equal(thumbPrints[file], thumbprint);
+            }
+        }
+
+        public Task InitializeAsync()
+        {
+            return PackageTestProjects.PackAsync(Output);
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+    }
+
+    public class PackageTestProjectsFixture
+    {
+        private bool _packed;
+
+        internal async Task PackAsync(ITestOutputHelper output)
+        {
+            if (_packed)
+            {
+                return;
+            }
+
+            var projectsToPack = GetProjectsToPack();
+
+            foreach (var project in projectsToPack)
+            {
+                output.WriteLine(project);
+            }
+
+            foreach (var project in projectsToPack)
+            {
+                var psi = new ProcessStartInfo
+                {
+                    FileName = DotNetMuxer.MuxerPathOrDefault(),
+#if DEBUG
+                    Arguments = "pack -c Debug",
+#else
+                    Arguments = "pack -c Release",
+#endif
+                    WorkingDirectory = project,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                };
+                var tcs = new TaskCompletionSource<Process>();
+                var process = Process.Start(psi);
+                process.Exited += (s, a) =>
+                {
+                    tcs.SetResult(process);
+                };
+                process.EnableRaisingEvents = true;
+
+                await Task.WhenAny(Task.Delay(TimeSpan.FromMinutes(2)), tcs.Task);
+
+                // Wait for 10 seconds or fail. If we take longer
+                // it likely means we are adding unexpected dependencies.
+                Assert.True(process.HasExited, "Process did not finish running.");
+                output.WriteLine(process.StandardOutput.ReadToEnd());
+                Assert.Equal(0, process.ExitCode);
+            }
+
+            _packed = true;
+        }
+
+        public static string[] GetProjectsToPack()
+        {
+            return typeof(PackageTestProjectsFixture).Assembly.GetCustomAttributes<AssemblyMetadataAttribute>()
```

Comment:
Just use a bespoke list. This indirection isn't helpful.

Reply:
changed